### PR TITLE
Reorganize FastSim Run3 workflow

### PR DESCRIPTION
**Pull request by srimanob**
#### PR description:
This PR is to reorganize the Run3 FastSim workflow.

Before: 
- 11634.0 (FullSim) ==> 11634.301 (FastSim) seems to be easy, but it needs several hacks if we would like to make it works for PU, Premixing test
- With this PR, FastSim has its own key in upgrade. So it needs less hacks to handles several PU workflows.

To be done:
- Make the Premixing workflow for FastSim.

Comments are welcomed.

#### PR validation:
Running with configs out from the following runTheMatrix:
`runTheMatrix.py --what upgrade -l 13234.0,13434.0,13240.303 --wm init`

Comparison of workflow list can be done with
`diff /afs/cern.ch/user/s/srimanob/public/ForFastSim/runTheMatrix_original.log /afs/cern.ch/user/s/srimanob/public/ForFastSim/runTheMatrix_withPR_v2.log`
It shows that previous `.301, .302, .303` are removed, and new workflows are added, started from `13200.`.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:
No need of backport.

### CODE REVIEW

**Comment by srimanob** [comment]
test parameters:
- workflows = 13234.0,13434.0,13240.303

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by kpedro88** [comment]
@srimanob have you `diff`ed the output of `runTheMatrix.py -w upgrade -n` before and after this modification to ensure that only the intended workflows were added and no other changes occurred? (It's unfortunate to have to make so many manual modifications, but the need to use different pileup inputs seems to make this the easiest path.)

**Comment by srimanob** [comment]
Hi @kpedro88 
Difference can seen with
`diff /afs/cern.ch/user/s/srimanob/public/ForFastSim/runTheMatrix_original.log /afs/cern.ch/user/s/srimanob/public/ForFastSim/runTheMatrix_withPR.log`

It shows that previous `.301, .302, .303` are removed, and new workflows are added, started from `13200.` This PR adds more workflows than needed, i.e. it include `.5XX` (Patatrack) also.

**Comment by srimanob** [comment]
FYI @sbein

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by srimanob** [comment]
Note to @cms-sw/pdmv-l2 on new workflows for fastsim run3 validation

NoPU
`13234.0,13207.0,13208.0,13246.0,13292.0,13243.0,13252.0,13250.0,13330.0`
PU (Classical mixing)
`13434.0,13407.0,13408.0,13446.0,13492.0,13443.0,13452.0,13450.0,13530.0`

TTbar: 13234.0 (2021FS+TTbar_14TeV_TuneCP5_FastSimRun3+HARVESTFastRun3)
SingleMuPt10: 13207.0 (2021FS+SingleMuPt10_Eta2p85_FastSimRun3+HARVESTFastRun3)
SingleMuPt100: 13208.0 (2021FS+SingleMuPt100_Eta2p85_FastSimRun3+HARVESTFastRun3)
ZEE: 13246.0 (2021FS+ZEE_14TeV_TuneCP5_FastSimRun3+HARVESTFastRun3)
ZTT: 13292.0 (2021FS+ZTT_All_hadronic_14TeV_TuneCP5_FastSimRun3+HARVESTFastRun3)
QCD_FlatPt_15_3000: 13243.0 (2021FS+QCDForPF_14TeV_TuneCP5_FastSimRun3+HARVESTFastRun3)
H125GGgluonfusion: 13252.0 (2021FS+H125GGgluonfusion_14TeV_TuneCP5_FastSimRun3+HARVESTFastRun3)
ZMM: 13250.0 (2021FS+ZMM_14TeV_TuneCP5_FastSimRun3+HARVESTFastRun3)
SMS-T1tttt_mGl-1500_mLSP-100: 13330.0 (2021FS+SMS-T1tttt_mGl-1500_mLSP-100_TuneCP5_14TeV_pythia8_FastSimRun3+HARVESTFastRun3)

**Comment by srimanob** [comment]
@cmsbuild please test

To refresh the test

**Comment by sbein** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

Presented in the SIM meeting, https://indico.cern.ch/event/1182398/
I prefer this PR to go first, to make sure that overall is OK, including IB test. Then I will need to rebase https://github.com/cms-sw/cmssw/pull/38736 and also update the Nano validation.

**Comment by srimanob** [comment]
@cms-sw/pdmv-l2 
Could you please review this PR? I will then rebase another PR on the miniaod + nanoaod validations for FastSim. Thanks.
The summary is of all, including this PR is at https://indico.cern.ch/event/1182398/contributions/4967724/attachments/2480443/4258755/20220715-Workflows.pdf

**Comment by kskovpen** [comment]
Thanks a lot for all the updates @srimanob . I wonder if jenkins tests would catch all the existing wfs that could be potentially touched by these changes (I haven't seen any problems). Otherwise, it is fine with us.

**Comment by srimanob** [comment]
@perrotta @qliphy 
Not sure if you have comment on this PR before merging. I can't join the release meeting today. I can follow up later.

The plan is to merge this PR first. Let's IB done, and see nothing wrong. Then I will rebase https://github.com/cms-sw/cmssw/pull/38736 to support MiniAOD and NanoAOD DQM, and some structure change for FastSim.

**Comment by srimanob** [comment]
Anthing that stops to merge this PR?

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 1]
```diff
@@ -2078,14 +2043,25 @@ def condition(self, fragment, stepList, key, hasHarvest):
         'BeamSpot': 'Run3RoundOptics25ns13TeVLowSigmaZ',
         'ScenToRun' : ['GenSim','Digi','RecoNano','HARVESTNano','ALCA'],
     },
+    '2021FS' : {
+        'Geom' : 'DB:Extended',
+        'GT' : 'auto:phase1_2022_realistic',
+        'HLTmenu': '@relval2022',
+        'Era' : 'Run3_FastSim',
+        'BeamSpot': 'Run3RoundOptics25ns13TeVLowSigmaZ',
```
[the remark this answers is not in the document]
Comment (qliphy) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2051]: I am fine with both options.  maybe @perrotta has preference?

[the remark this answers is not in the document]
Comment (srimanob) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2051]: Hi @qliphy 
As those PR is still in progress. I prefer to merge this first. The next PR will come in 2-3 days. Thanks.

**Comment by perrotta** [comment]
+1
- Let update all beam spot coherently in #38760

**Pull request merged**